system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -41,67 +43,136 @@ extern "C" MetricSink* getSinkInstance(const char *name, const IPropertyTree *pS
 ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSettingsTree) :
     PeriodicMetricSink(name, "elastic", pSettingsTree)
 {
+    // Standard sink settings
     ignoreZeroMetrics = pSettingsTree->getPropBool("@ignoreZeroMetrics", true);
 
-    StringBuffer hostName;
+    // Get the host and index configuration
+    if (getHostConfig(pSettingsTree) && getIndexConfig(pSettingsTree))
+    {
+        configurationValid = true;
+        PROGLOG("ElasticMetricSink: Loaded and configured");
+    }
+}
+
+
+bool ElasticMetricSink::getHostConfig(const IPropertyTree *pSettingsTree)
+{
+    StringBuffer hostDomain;
     StringBuffer hostProtocol;
     StringBuffer hostPort;
 
     Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
     if (pHostConfigTree)
     {
-        pHostConfigTree->getProp("@name", hostName);
+        pHostConfigTree->getProp("@domain", hostDomain);
 
         if (!pHostConfigTree->getProp("@protocol", hostProtocol))
-        {
             hostProtocol.append("https");
-        }
 
         if (!pHostConfigTree->getProp("@port", hostPort))
-        {
             hostPort.append("9200");
-        }
     }
 
-    if (!hostName.isEmpty() && !hostPort.isEmpty() && !hostProtocol.isEmpty())
+    // Validate the host configuration minimal settings are present
+    if (hostDomain.isEmpty() || hostProtocol.isEmpty())
     {
-        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+        WARNLOG("ElasticMetricSink: Host configuration missing domain and/or protocol");
+        return false;
     }
-    else
+
+    // build url for use with httplib Client
+    elasticHostUrl.append(hostProtocol).append("://").append(hostDomain);
+    if (!hostPort.isEmpty())
+        elasticHostUrl.append(":").append(hostPort);
+
+    // Read optional certificate file path
+    pHostConfigTree->getProp("@certificateFilePath", certificateFilePath);
+
+    // Get authentication settings, if present
+    Owned<IPropertyTree> pAuthConfigTree = pSettingsTree->getPropTree("authentication");
+    if (pAuthConfigTree)
```

Comment:
if !pAuthConfigTree could be handled the same way pAuthConfigTree handles it in L153

Reply:
I changed and pushed a new commit. However, if the auth config tree is missing, true is returned, not false since it's not an error.